include/cmdlib/CommandFacility.hpp

```diff
@@ -66,16 +66,16 @@ class CommandFacility
   virtual void run(std::atomic<bool>& end_marker) = 0;
 
   //! Feed commands from the implementation.
-  void execute_command(const cmdobj_t& command);
+  void execute_command(cmdmeta_t meta);
```

Comment:
I would keep the `command` object immutable. Also the parameter being called meta hides that fact that the command itself is encapsulated in it.

Reply:
I put back the immutable `cmdobj_t` and put `cmdmeta_t` as second (mutable) parameter